src/patches/third-party-main.js

```diff
@@ -0,0 +1,57 @@
+export default async function main (compiler, next) {
+  const mainFile = await compiler.readFileAsync('lib/_third_party_main.js')
+  mainFile.contents = `
+Object.defineProperty(process, '__nexe', (function () {
+  let nexeHeader = null
+  return {
+    get: function () {
+      return nexeHeader
+    },
+    set: function (value) {
+      if (nexeHeader) {
+        throw new Error('__nexe cannot be reconfigured')
+      }
+      nexeHeader = value
+      Object.freeze(nexeHeader)
+    },
+    enumerable: false,
+    configurable: false
+  }
+})());
+
+const fs = require('fs')
+const originalReadFile = fs.readFile
+const Buffer = require('buffer').Buffer
+const isString = x => typeof x === 'string' || x instanceof String
```

Comment:
this is in `_third_party_main.js` (a common js module) so require will wrap it like it does every other module.